BJS2-69961. Выполнение задачи "Lord of the Rings RPG"

Keriod opened the pull request.

Keriod committed:
BJS2-69961. PR corrections

20cognosce commented on src/main/java/school/faang/lambda/lotr_rpg/InventoryManager.java:
```diff
@@ -0,0 +1,24 @@
+    }
+
+    public void updateItem(Character character, Predicate<Item> condition, Function<Item, Item> updater) {
+        character.inventory().forEach(item -> {
+            if (condition.test(item)) {
+                character.inventory().set(character.inventory().indexOf(item), updater.apply(item));
+            }
+        });
```
Допустим это окей, но ты изменяешь коллекцию, по которой сам в этот же момент итерируешься, это не очень хорошо
и if (condition.test) - топорно очень как-то)

Давай переделаем более красиво
Тебе понадобится сделать два стрима:
один c фильтром по condition + применением функции через определенный метод стрима
второй - с фильтром по отрицанию condition

Затем соединить два стрима в один (Stream имеет такой метод) и обновить весь инвентарь целиком

20cognosce commented on src/main/java/school/faang/lambda/lotr_rpg/Main.java:
```diff
@@ -0,0 +1,22 @@
+        manager.removeItem(frodo, (item) -> item.name().contains("Ring"));
+
+        manager.addItem(frodo, ring, (item) -> System.out.println(item.name() + " снова добавлен."));
+        manager.updateItem(frodo, (item) -> item.name().equals("The One Ring"),
```
не вызывай никогда equals на объектах, они могут быть null
используй Objects.equals(a, b)

Keriod committed:
BJS2-69961.
- Changed InventoryManager.updateItem to perform with two streams concatenated into one
and rewriting character inventory
- Changed Character from record to class
- Added test in Main

20cognosce commented on src/main/java/school/faang/lambda/lotr_rpg/Character.java:
```diff
@@ -0,0 +1,15 @@
+@AllArgsConstructor
+public class Character {
+    private final String name;
+    @Setter
+    private List<Item> inventory;
```
Нет, он final

20cognosce commented on src/main/java/school/faang/lambda/lotr_rpg/InventoryManager.java:
```diff
@@ -0,0 +1,32 @@
+        Stream<Item> falseCondition = character
+                .getInventory()
+                .stream()
+                .filter(item -> !condition.test(item));
```
Есть метод лучше

Keriod replied:
condition.negate?

20cognosce commented on src/main/java/school/faang/lambda/lotr_rpg/InventoryManager.java:
```diff
@@ -0,0 +1,32 @@
+                .stream()
+                .filter(item -> !condition.test(item));
+
+        character.setInventory(Stream.concat(trueCondition, falseCondition).toList());
```
Без сеттера

Keriod committed:
BJS2-69961.
- PR changes
